### ISSUE TITLE
Refactor to Consolidate `Group` and `User` into Single `Account` Object

### ISSUE THREAD

**Issue by mufniarz** [ASCENDynamics-NFP/AscendCoopPlatform#134]
**Context:**
Currently, our platform's architecture differentiates between `User` and `Group` objects, as detailed in our [User Schema](https://github.com/ASCENDynamics-NFP/AscendCoopPlatform/wiki/User-Schema) and [Group Schema](https://github.com/ASCENDynamics-NFP/AscendCoopPlatform/wiki/Group-Schema) documentation. While this separation has served us well, we've identified several areas where a unified model could streamline processes and reduce complexity, particularly in the context of our [Relationships Schema](https://github.com/ASCENDynamics-NFP/AscendCoopPlatform/wiki/Relationships-Schema).

**Proposal:**
Refactor the application to consolidate the `Group` and `User` objects into a single `Account` object. This new `Account` model will encapsulate the functionalities and attributes of both users and groups, thereby simplifying our data model and interactions between different types of accounts.

**Goals:**
- Simplify the data model to enhance maintainability and scalability.
- Reduce redundancy in code where `User` and `Group` functionalities overlap.
- Streamline interactions and relationships between users and groups.
- Facilitate easier implementation of features that apply to both users and groups.

**Proposed Changes:**
1. **Define a New `Account` Schema:**
   - The `Account` schema should include common fields from both `User` and `Group` schemas.
   - Identify and integrate unique attributes of `User` and `Group` into the `Account` schema appropriately.

2. **Update Relationship Management:**
   - Refactor the [Relationships Schema](https://github.com/ASCENDynamics-NFP/AscendCoopPlatform/wiki/Relationships-Schema) to accommodate the new `Account` model.
   - Ensure that existing relationships (e.g., group memberships, user connections) are preserved and accurately represented in the new model.

3. **Data Migration Strategy:**
   - Develop a comprehensive strategy for migrating existing `User` and `Group` data to the new `Account` model.
   - Ensure data integrity and minimal downtime during the migration process.

4. **Update Application Logic:**
   - Refactor existing application logic to interact with the new `Account` model.
   - Pay special attention to areas where `User` and `Group` logic diverges, ensuring that these are appropriately handled in the unified model.

5. **Testing and Validation:**
   - Thoroughly test the refactored application to ensure all functionalities are working as expected.
   - Include both automated and manual testing, focusing on data integrity, feature parity, and user experience.

**Impact Analysis:**
- Discuss potential impacts on existing features and user experience.
- Outline any anticipated challenges or limitations introduced by this change.

**Request for Comments:**
I invite team members to provide feedback on this proposal, particularly regarding:
- Potential challenges and how we might address them.
- Additional benefits this consolidation might offer.
- Insights on ensuring a smooth transition for our users during the migration to the new model.

Looking forward to the team's input and collaboration on this significant refactor.

**Comment by AMIL-C** [comment]
Hey Marcin,

Hope you're doing well these days. I know it's been a while. Just the same,
are you still working with JavaScript. Been trying to learn the language on
my own, and I'm having trouble getting the hang of it.

Amil

**Comment by malsmr** [comment]
Is this ready to be picked up?

**Comment by mufniarz** [comment]
Hi @malsmr, I've actually started work on this.  It's a bit of work, as when I replaced the collections with Account.  It also made me realize that I should just use a sub-collection for relationships.  Which added to the refactoring work.  I'm close to done, but it's going slowly.  I've needed to touch almost every file, if not every file, but I think once I'm done with this it should simplify the code and database. Let me know if you have any more questions, I'm happy to connect on the progress and see if there's a place were others can jump in.
